### PR TITLE
[Add] Add homework 4

### PR DESCRIPTION
**Pull request by AngelStoyanov33**

### CODE REVIEW

File: .github/workflows/homework_04_02_angel_stoyanov.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
+name: Homework 04 - 02 Angel Stoyanov
```
Comment (Teodor1331) [inline .github/workflows/homework_04_02_angel_stoyanov.yaml:1]: The only problem I see in the YAML file is the convention. Your names should start with lower case - here and below on lines 6, 12, 37 and 42. Everything else here seems okay for me. Excellent!

File: .github/workflows/homework_04_02_angel_stoyanov.yaml [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+          cd homework_04/02_Angel_Stoyanov
+          pytest
+        
+      - name: Run Pylint
+        run: |
+          cd homework_04/02_Angel_Stoyanov
+          pylint circle.py
+          pylint point.py
+          pylint relative_point.py
+          pylint test_cirlce.py
```
Comment (Teodor1331) [inline .github/workflows/homework_04_02_angel_stoyanov.yaml]: You can simplify it a little with pylint *.py, but this way is good too here for just 4 files.

File: homework_04/02_Angel_Stoyanov/point.py [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        """A setter method for the y coordinate field"""
+        if isinstance(new_y, float) is not True:
+            raise ValueError("Y should be of type float")
+        self.__y = new_y
+
+    @staticmethod
+    def get_distance_between_two_points(point_a, point_b):
+        """A static method that returns the distance between two points"""
+        return sqrt(pow((point_a.get_x() - point_b.get_x()), 2) + \
+            pow((point_a.get_y() - point_b.get_y()), 2))
```
Comment (Teodor1331) [inline homework_04/02_Angel_Stoyanov/point.py:35]: Here you can simplify it by using the exponential operator ** with exponent of 2 or 0.5, but this way is very good too, no objections. I like that you have made the method a static one.

**Review by Teodor1331 — APPROVED** [review]
You have done correctly the homework according to me! Good job!

File: .github/workflows/homework_04_02_angel_stoyanov.yaml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+      - homework_04/02_Angel_Stoyanov/**
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: ['3.8', '3.9', '3.10']
```
Comment (Martincho2003) [inline .github/workflows/homework_04_02_angel_stoyanov.yaml:21]: nice for testing in different python versions

**Review by Martincho2003 — APPROVED** [review]
Since the pylon and the pytest are successful, everything is fine.

**Pull request merged**